src/writers/BaseWriter.ts

```diff
@@ -928,7 +928,7 @@ export abstract class BaseWriter<T extends BaseWriterOptions, U extends XMLSeria
      * 5. Replace any occurrences of ">" in markup by "&gt;".
      * 6. Return the value of markup.
      */
-    const markup = node.data.replace(/(?!&([^&; ]*);)&/g, '&amp;')
+    const markup = node.data.replace(/&/g, '&amp;')
```

Comment:
FYI, this fix will double encode already html encoded entities `&amp;` => `&amp;amp;`

Reply:
Probably better to fix the regex with a negative lookahead `/&(?!amp;)/g`